gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineNullAway.java

```diff
@@ -90,4 +101,38 @@ public void execute(JavaCompile javaCompile) {
             }
         });
     }
+
+    // Workaround for nullaway bugs in the last release to support jdk-15. Projects that don't use jdk-15
+    // resolve a newer nullaway version with bug-fixes. This may be deleted after we've rolled everything
+    // off jdk-15 MTS.
+    private static void newerNullAwayInNonJdk15Projects(Project project) {
+        project.getConfigurations()
+                .matching(new Spec<Configuration>() {
+                    @Override
+                    public boolean isSatisfiedBy(Configuration config) {
+                        return "errorprone".equals(config.getName());
+                    }
+                })
+                .configureEach(new Action<Configuration>() {
+                    @Override
+                    public void execute(Configuration _files) {
+                        for (String dep : NON_15_DEPS) {
+                            project.getDependencies().addProvider("errorprone", project.provider(() -> {
+                                // Cannot upgrade dependencies in this case because newer nullaway/checkerframework
+                                // are not compatible with java 15, but fully support jdk11 and jdk17.
+                                return anyProjectUsesJava15(project) ? null : dep;
```

Comment:
I wonder if this should be the other way around? Per default, we use the latest versions but downgrade to the old compatible version when using j15? That way, j15 is the special case and usage of this hack will phase out naturally once projects roll to j17?

Reply:
How would you suggest doing that? Part of the issue is that it would bump errorprone deps as well to a version that’s no longer compatible with jdk15. Gcv makes upgrades much easier than downgrades :-/

Reply:
Hmm yeah, I was thinking of using `resolutionStrategy.force("java-15-dep-coordinate")` for all deps that we need to pin to a specific version when using j15. But yeah, we would have to enumerate all the error-prone deps that are also not compatible and this might cause issues when other deps depend on them.

Reply:
Hmm yeah, this is becoming more complicated. I guess its fine to put this in for now, given that jdk 15 is EOL soon and we can remove the special casing.